taichi/ir/ir.h

```diff
@@ -710,7 +722,11 @@ class RandStmt : public Stmt {
     TI_STMT_REG_FIELDS;
   }
 
-  virtual bool has_global_side_effect() const override {
+  bool has_global_side_effect() const override {
+    return false;
```

Comment:
I think `RandStmt` has global side effect? i.e. change random seed.

Reply:
I think we can assume random number generator calls are independent from each other. Therefore. although `RandStmt` changes RNG states, it should not be considered visible to other statements.